lib/ModuleWarning.js

```diff
@@ -2,15 +2,20 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-function ModuleWarning(module, warning) {
-	Error.call(this);
-	Error.captureStackTrace(this, ModuleWarning);
-	this.name = "ModuleWarning";
-	this.module = module;
-	this.message = warning;
-	this.warning = warning;
+"use strict";
+
+class ModuleWarning extends Error {
+	constructor(module, warning) {
+		super();
+
+		if(Error.hasOwnProperty("captureStackTrace")) {
+			Error.captureStackTrace(this, ModuleWarning);
```

Comment:
Same comment as [here](https://github.com/webpack/webpack/pull/3853#discussion_r95105601). If you guys are confortable with these changes I can update the PR without problems.